examples/basf2/basf2_chain_example.py

```diff
@@ -23,23 +23,22 @@ class SimulationTask(Basf2PathTask):
     event_type = luigi.EnumParameter(enum=SimulationType)
 
     def create_path(self):
-        path = basf2.create_path()
+        path = basf2.Path()
         modularAnalysis.setupEventInfo(self.n_events, path)
 
         if self.event_type == SimulationType.y4s:
             # in current main branch and release 5 the Y(4S)decay file is moved, so try old and new locations
             find_file_ignore_error = True
-            dec_file = Belle2.FileSystem.findFile('analysis/examples/tutorials/B2A101-Y4SEventGeneration.dec',
-                                                  find_file_ignore_error)
-            if not dec_file:
-                dec_file = Belle2.FileSystem.findFile('analysis/examples/simulations/B2A101-Y4SEventGeneration.dec')
+            dec_file = basf2.find_file('analysis/examples/tutorials/B2A101-Y4SEventGeneration.dec',
+                                       silent=find_file_ignore_error)
+            if dec_file == '':
```

Comment:
Why explicitly check for an empty string? I assume that's what `basf2.find_file` returns if no file was found? Empty strings also evaluate to False in Python, I'm sure you know. I'm not saying this is wrong, just want to understand the reason.

I'm not sure if there are cases where it returns something other that evaluates to `False` like `None`. Probably not. But maybe we can be paranoid and raise a `ValueError` or `TypeError`  if it returns something other that was covered by the `not` but not by the `== ''`.

Reply:
With `silent=True`, `find_file` returns `''` if nothing is found. With `silent=False`, a `FileNotFoundError` exception is raised.

Why did I use `== ''`? Because I don't know Python. :D

Reply:
Thanks, I reverted the empty string check to `if not dec_file` in b1ff807c9199d6decd560a89dd91f12bd34a5632 and added your explanation from this comment as a code-comment.